packages/contracts-periphery/contracts/universal/op-nft/Optimist.sol

```diff
@@ -0,0 +1,135 @@
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./AttestationStation.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract Optimist is
+    Initializable,
+    ERC721Upgradeable,
+    ERC721BurnableUpgradeable,
+    OwnableUpgradeable,
+    Semver
+{
+    AttestationStation public attestationStation;
+    address public admin;
+
+    /**
+     * TODO do I initialize?
+     */
+    constructor() Semver(0, 0, 1) {}
+
+    /**
+     * @notice  Initialize the Optimist contract.
+     * @dev     Initializes the Optimist contract with the given parameters.
+     * @param   _name  The token name.
+     * @param   _symbol  The token symbol.
+     * @param   _admin  The administrator address.
+     * @param   _attestationStation  The address of the attestation station contract.
+     */
+    function initialize(
+        string calldata _name,
+        string calldata _symbol,
+        address _admin,
+        address _attestationStation
+    ) public initializer {
+        __ERC721_init(_name, _symbol);
+        __ERC721Burnable_init();
+        __Ownable_init();
+        attestationStation = AttestationStation(_attestationStation);
+        admin = _admin;
+    }
+
+    /**
+     * @notice  Mint the Optimist token.
+     * @dev     Mints the Optimist token to the give recipient address.
+     *          Limits the number of tokens that can be minted to one per address.
+     *          The tokenId is the uint256 of the recipient address.
+     * @param   _recipient  The address of the token recipient.
+     */
+    function mint(address _recipient) public {
+        require(balanceOf(_recipient) == 0, "Optimist::mint: ALREADY_MINTED");
+        require(isWhitelisted(_recipient), "Optimist::mint: NOT_WHITELISTED");
+        uint256 tokenId = uint256(uint160(_recipient));
+        _safeMint(_recipient, tokenId);
+    }
+
+    /**
+     * @notice Returns the URI for the token metadata.
+     * @dev The token URI will be stored at baseURI + '/' + tokenId + .json
+     * @param tokenId The token ID to query.
+     * @return The URI for the given token ID.
+     */
+    function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
+        if (ownerOf(tokenId) == address(0)) {
+            revert("Optimist:::tokenURI: TOKEN_URI_DNE");
+        }
+        return string(abi.encodePacked(baseURI(), "/", Strings.toHexString(tokenId), ".json"));
+    }
+
+    /**
+     * @notice  Returns whether an address is whitelisted
+     * @dev     The whitelist is an attestation by the admin of this contract
+     * @return  boolean  Whether the address is whitelisted
+     */
+    function isWhitelisted(address _recipient) public view returns (bool) {
```

Comment:
rename canMint?

Reply:
Would prefer not to rename since it's a breaking change to web app

Reply:
Wouldn't be opposed to renaming this

Reply:
I'd prefer renaming, `whitelist` is a dirty word